assets/js/modules/tagmanager/components/setup/SetupErrorNotice.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * Tag Manager Setup Error Notice component.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { STORE_NAME } from '../../datastore/constants';
+import { STORE_NAME as MODULES_ANALYTICS } from '../../../analytics/datastore/constants';
+import { STORE_NAME as CORE_MODULES } from '../../../../googlesitekit/modules/datastore/constants';
+import StoreErrorNotice from '../../../../components/StoreErrorNotice';
+import ErrorText from '../../../../components/error-text';
+const { useSelect } = Data;
+
+export default function SetupErrorNotice() {
+	const analyticsErrors = [
+		// Check if activating Analytics failed.
+		useSelect( ( select ) => select( CORE_MODULES ).getErrorForAction( 'activateModule', [ 'analytics' ] ) ),
+		// Check if saving Analytics settings failed.
+		useSelect( ( select ) => {
+			const settings = select( MODULES_ANALYTICS ).getSettings();
+			return select( MODULES_ANALYTICS ).getErrorForAction( 'saveSettings', [ settings ] );
+		} ),
+	].filter( Boolean );
+
+	if ( analyticsErrors.length ) {
+		return analyticsErrors.map(
+			( { message, reconnectURL } ) => <ErrorText key={ message } message={ message } reconnectURL={ reconnectURL } />
+		);
```

Comment:
Is it okay to return an array from a component like this? I'd think you'd want to wrap this in a `<Fragment>` or the return value of this component will be an `Array`. I thought only components, strings, `false`, or `null` were valid return types.

```suggestion
		return (
			<Fragment>
				{analyticsErrors.map( ( { message, reconnectURL } ) => <ErrorText key={ message } message={ message } reconnectURL={ reconnectURL } />}
			</Fragment>
		);
```

seems like it'd do the trick.

Reply:
I had my doubts about this too but it [seems to work just fine](https://codesandbox.io/s/condescending-thompson-8yyw1?file=/src/App.js) 😄 

Apparently this was added in React v16:
https://reactjs.org/blog/2017/09/26/react-v16.0.html#new-render-return-types-fragments-and-strings